llvm/test/CodeGen/AMDGPU/rotl.ll

```diff
@@ -25,24 +25,28 @@ define amdgpu_kernel void @rotl_i32(ptr addrspace(1) %in, i32 %x, i32 %y) {
 ; SI-NEXT:    s_load_dwordx4 s[0:3], s[4:5], 0x9
 ; SI-NEXT:    s_mov_b32 s7, 0xf000
 ; SI-NEXT:    s_waitcnt lgkmcnt(0)
-; SI-NEXT:    s_sub_i32 s3, 32, s3
+; SI-NEXT:    s_sub_i32 s4, 32, s3
```

Comment:
This looks worse but I suppose it's just the consequence of now being scalar (though in this type of case we probably should have rewritten the tail to fold into the copy to VGPR)